### PR TITLE
Isolate the code for SimpleXmlReader parser

### PR DESCRIPTION
**Pull request by ayushsingh01042003**
# Description

- Some code restructuring and cleanup
- Moving the static functions to the EDStatic Class
- Functions for sending particular emails

### CODE REVIEW

File: WEB-INF/classes/gov/noaa/pfel/erddap/util/EDStatic.java [hunk 1]
```diff
@@ -6763,4 +6763,92 @@ else if (errorNo == HttpServletResponse.SC_SERVICE_UNAVAILABLE) //http error 503
...
+        EDStatic.tally.remove("tabledap DatasetID (since last daily report)");
+        EDStatic.tally.remove("tabledap File Type (since last daily report)");
+        EDStatic.tally.remove("WCS index.html (since last daily report)");
+        EDStatic.tally.remove("WMS doWmsGetMap (since last daily report)");
+        EDStatic.tally.remove("WMS doWmsGetCapabilities (since last daily report)");
+        EDStatic.tally.remove("WMS doWmsDemo (since last daily report)");
+        EDStatic.tally.remove("WMS index.html (since last daily report)");
+    }
+
+    public static void resetTimeDistributions() {
```
Comment (ChrisJohnNOAA) [inline WEB-INF/classes/gov/noaa/pfel/erddap/util/EDStatic.java]: resetDailyDistributions

This doesn't reset all of the distributions on EDStatic

File: WEB-INF/classes/gov/noaa/pfel/erddap/util/EDStatic.java [hunk 2]
```diff
@@ -6763,4 +6763,92 @@ else if (errorNo == HttpServletResponse.SC_SERVICE_UNAVAILABLE) //http error 503
...
+        EDStatic.tally.remove("Requester's IP Address (Allowed) (since last Major LoadDatasets)");
+        EDStatic.tally.remove("Requester's IP Address (Blacklisted) (since last Major LoadDatasets)");
+        EDStatic.tally.remove("Requester's IP Address (Failed) (since last Major LoadDatasets)");
+        EDStatic.tally.remove("Requester's IP Address (Too Many Requests) (since last Major LoadDatasets)");
+
+        EDStatic.failureTimesDistributionLoadDatasets  = new int[String2.TimeDistributionSize];
+        EDStatic.responseTimesDistributionLoadDatasets = new int[String2.TimeDistributionSize];
+    }
+
+    public static void clearAllTallies() {
```
Comment (ChrisJohnNOAA) [inline WEB-INF/classes/gov/noaa/pfel/erddap/util/EDStatic.java]: clearDailyTallies

File: WEB-INF/classes/gov/noaa/pfel/erddap/LoadDatasets.java [hunk 3]
```diff
@@ -861,441 +1012,237 @@ public void run() {
...
+                        subject + ": see log.txt for details.\n";  //swap into place
+                String content = MustBe.throwableToString(t);
+                unexpectedError = subject + ": " + content;
+                String2.log(unexpectedError);
+                EDStatic.email(EDStatic.emailEverythingToCsv, subject, content);
             }
+        } finally {
+            if (xmlReader != null)
+                try {xmlReader.close();} catch (Exception e) {}
+            EDStatic.suggestAddFillValueCSV.setLength(0);
```
Comment (ChrisJohnNOAA) [inline WEB-INF/classes/gov/noaa/pfel/erddap/LoadDatasets.java]: I believe `EDStatic.suggestAddFillValueCSV` is used as part of the `handleAfva`. I think the call `EDStatic.suggestAddFillValueCSV.setLength(0);` is unrelated to the xml parsing function and so should be in the main run.

File: WEB-INF/classes/gov/noaa/pfel/erddap/LoadDatasets.java [hunk 4]
```diff
@@ -861,441 +1012,237 @@ public void run() {
...
-                "  nDatasets in datasets.xml=" + nDatasets + " (nTry=" + nTry + ")\n" +
-                "  nUsers=" + tUserHashMap.size());
-
-            //minorLoad?
-            if (!majorLoad) {
-                String2.distributeTime(loadDatasetsTime, EDStatic.minorLoadDatasetsDistribution24);
-                String2.distributeTime(loadDatasetsTime, EDStatic.minorLoadDatasetsDistributionTotal);
-                String2.log(datasetsThatFailedToLoad);
+        } catch (Throwable t) {
+            String subject = String2.ERROR + " while processing " +
```
Comment (ChrisJohnNOAA) [inline WEB-INF/classes/gov/noaa/pfel/erddap/LoadDatasets.java]: This is minor, but we should move this string definition into the if.

**Review by ChrisJohnNOAA — CHANGES_REQUESTED** [review]
A few small changes, mostly naming.

**Comment by ayushsingh01042003** [comment]
Does this look good?

**Review by ChrisJohnNOAA — APPROVED** [review]

**Pull request merged**